src/util/AbstractMemory/MemoryDefinitionLanguageVisitor.h

```diff
@@ -0,0 +1,27 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (July of 2023, schlegea@informatik.uni-freiburg.de)
+#pragma once
+
+#include <antlr4-runtime.h>
+
+#include "util/AbstractMemory/generated/MemoryDefinitionLanguageParser.h"
+#include "util/json.h"
+
+/*
+This visitor will translate the memory definition language to the amount of
+bytes defined.
+*/
+class ToSizeTMemoryDefinitionLanguageVisitor final {
+ public:
+  using Parser = MemoryDefinitionLanguageParser;
+
+  size_t visitMemoryDefinitionString(
+      Parser::MemoryDefinitionStringContext* context) const;
+
+  size_t visitPureByteDefinition(
+      Parser::PureByteDefinitionContext* context) const;
+
+  size_t visitMemoryUnitDefinition(
+      Parser::MemoryUnitDefinitionContext* context) const;
```

Comment:
Most of them can be private, and the public "main" function should return a memory object directly.
That way this whole thing gets more expressive and more typesafe.

Reply:
Done, see [53f9f78](https://github.com/ad-freiburg/qlever/pull/1032/commits/53f9f7807b8559f4f6588e8e9f33ac9b4467c379).